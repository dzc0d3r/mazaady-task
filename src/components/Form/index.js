import Form from "./Form"
import Loading from "./Loading"

export { Form, Loading }
