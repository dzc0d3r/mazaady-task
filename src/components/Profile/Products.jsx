import { FiPlusCircle } from "react-icons/fi"
import { SingleProduct } from "@/components/Profile"

const Products = ({ products }) => {
  return (
    <div className="h-full w-full rounded-2xl bg-white p-5">
      <div className="flex flex-col items-start justify-between gap-2 md:flex-row md:items-center">
        <div className="flex flex-row gap-2">
          <button className="flex flex-row items-center gap-x-2 rounded-2xl border border-orange-500 bg-orange-200 bg-opacity-40 px-2 py-1 text-sm font-semibold text-orange-500 hover:to-orange-500 disabled:pointer-events-none disabled:opacity-50 md:px-5 md:py-2 md:text-base">
            Products
          </button>
          <button className="flex flex-row items-center gap-x-2 rounded-2xl border border-gray-300 bg-white/30 px-2 py-1 text-sm font-semibold text-gray-500 disabled:pointer-events-none disabled:opacity-50 md:px-5  md:py-2 md:text-base">
            Articles
          </button>
          <button className="flex flex-row items-center gap-x-2 rounded-2xl border border-gray-300 bg-white/30 px-2 py-1 text-sm font-semibold text-gray-500 disabled:pointer-events-none disabled:opacity-50 md:px-5  md:py-2 md:text-base">
            Reviews
          </button>
        </div>
        <button className="flex flex-row items-center gap-x-2 rounded-xl bg-gradient-to-r from-primary to-orange-600 px-2 py-2 text-sm font-semibold text-white hover:to-orange-500  disabled:pointer-events-none disabled:opacity-50 md:px-5 md:py-2 md:text-base">
          <FiPlusCircle className="text-lg md:text-2xl" />
          Add Review
        </button>
      </div>

      <div className="relative my-5 flex flex-col gap-2">
        <h1 className=" text-3xl font-semibold text-black/80 md:text-4xl ">
          Products
          <span className="absolute top-0 mx-3 text-base font-light  text-gray-700">
            (12)
          </span>
        </h1>
        <SingleProduct products={products} />
      </div>
    </div>
  )
}

export default Products
