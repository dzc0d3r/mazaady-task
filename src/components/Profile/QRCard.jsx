"use client"
import { IoEyeOutline } from "react-icons/io5"
import { GoShareAndroid } from "react-icons/go"
import { PiFileArrowDown } from "react-icons/pi"
import { TfiArrowCircleDown } from "react-icons/tfi"
import { useState } from "react"
import Image from "next/image"

const QRCard = () => {
  const [collapse, setCollapse] = useState(false)
  return (
    <div className="mx-auto flex h-fit w-full max-w-full flex-col rounded-2xl bg-white p-5 md:max-w-none">
      <div className="flex flex-row items-center justify-between">
        <h2 className="text-2xl font-semibold capitalize text-black/70">
          QR Code
        </h2>
        <div className="group flex flex-row gap-2">
          <IoEyeOutline className="text-2xl text-black/70 group-hover:cursor-pointer" />
          <GoShareAndroid className="text-2xl text-black/70 group-hover:cursor-pointer" />
          <PiFileArrowDown className="text-2xl text-black/70 group-hover:cursor-pointer" />
          <div className=" relative inline-flex lg:hidden">
            <input
              onClick={() => setCollapse(!collapse)}
              type="checkbox"
              id="faq1"
              className="peer absolute left-0 top-0 z-10 h-full w-full appearance-none text-center hover:cursor-pointer"
            />
            <TfiArrowCircleDown className="rounded-full bg-orange-200 bg-opacity-60 text-2xl text-black/70 " />
          </div>
        </div>
      </div>
      <div className={`${collapse ? "hidden" : "mt-4 flex flex-col"}`}>
        <div className="flex flex-row items-center gap-3 rounded-xl bg-orange-200 bg-opacity-40 p-5">
          <PiFileArrowDown className="text-3xl text-orange-400" />
          <p className="text-xs font-light text-black/70">
            Download the QR code or share it with your friends.
          </p>
        </div>
        <div className="mt-5 flex h-full w-full justify-center">
          <Image
            src={"qr.svg"}
            width={100}
            height={100}
            alt="Profile image"
            className="h-80 w-96 rounded-2xl"
          ></Image>
        </div>
      </div>
    </div>
  )
}

export default QRCard
