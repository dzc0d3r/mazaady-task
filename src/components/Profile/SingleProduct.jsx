import Image from "next/image"
import { PiHeart } from "react-icons/pi"
import { PiHeartFill } from "react-icons/pi"

const SingleProduct = ({ products }) => {
  return (
    <>
      {products.map((product, index) => (
        <div key={index} className="my-2 grid grid-cols-4 gap-4 p-0 md:p-4">
          <div className="group relative col-span-1 h-24 w-24 hover:cursor-pointer md:h-40 md:w-40">
            <Image
              src={product.imageSrc}
              fill
              alt="Profile image"
              className="absolute cursor-pointer rounded-2xl transition-all duration-300 group-hover:scale-[1.01] "
            ></Image>
            <div
              className={`-botton-px absolute right-0 flex md:-right-px md:bottom-[.55rem]  
                            ${product.tag.includes("live") ? "bg-primary" : "bg-orange-500"} h-fit w-fit rounded-ee-full rounded-ss-full px-2 py-[2px] text-[.6rem]  font-light capitalize text-white md:px-6 md:py-2 lg:text-sm`}
            >
              <p>{product.tag}</p>
            </div>
          </div>
          <div className="col-span-3 mx-7 mt-1 flex flex-row items-start md:mx-1 md:mt-4 md:justify-between">
            <div className="flex flex-col gap-3 ">
              <h3 className="max-w-52 break-words break-all text-sm font-medium text-black/70 lg:max-w-full lg:text-lg">
                {product.title}
              </h3>

              <p className="text-xs font-light text-black/70 md:mt-1 md:text-base ">
                Starting Price{" "}
                <span className="text-lg font-semibold text-black/80 md:text-2xl">
                  {product.starting_price} EGP
                </span>
              </p>
              <div className="flex flex-col items-start gap-2 md:flex-row md:items-center">
                <p className="flex items-center gap-2 text-xs font-light text-black/70 md:mt-1 md:flex-row md:text-base ">
                  Lot Starts In
                </p>
                <div className="flex flex-row gap-1 md:gap-2">
                  <span className="rounded-2xl bg-orange-200 px-2 py-[6px] text-xs font-semibold text-orange-500  md:px-4 md:py-1 md:text-xl">
                    {product.start_in.days}
                    <span className="mx-1 text-[.6rem] font-medium md:text-sm">
                      Days
                    </span>
                  </span>
                  <span className="rounded-2xl bg-orange-200 px-2 py-[6px] text-xs font-semibold text-orange-500  md:px-4 md:py-1 md:text-xl">
                    {product.start_in.hours}
                    <span className="mx-1 text-[.6rem] font-medium md:text-sm">
                      Hours
                    </span>
                  </span>
                  <span className="rounded-2xl bg-orange-200 px-2 py-[6px] text-xs font-semibold text-orange-500  md:px-4 md:py-1 md:text-xl">
                    {product.start_in.minutes}
                    <span className="mx-1 text-[.6rem] font-medium md:text-sm">
                      Minutes
                    </span>
                  </span>
                </div>
              </div>
            </div>
            {/* heart icon */}
            {product.liked ? (
              <PiHeartFill className="text-3xl text-primary transition-all duration-300 hover:scale-105 hover:cursor-pointer" />
            ) : (
              <PiHeart className="text-3xl text-gray-500 transition-all duration-300 hover:scale-105 hover:cursor-pointer" />
            )}
          </div>
        </div>
      ))}
    </>
  )
}

export default SingleProduct
