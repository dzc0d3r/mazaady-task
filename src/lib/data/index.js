const profileData = {
  imageSrc: "profile-card.svg",
  fullname: "Hala Ahmed",
  bio: "I am Hala Ahmed, I am the owner of the local brand called Beauty which is for Mackeup and Skin Care.",
  followers: 5,
  following: 10,
  rate: {
    value: 4.2,
    count: 15,
  },
  products: [
    {
      title: "Six-piece clothing set (blouse - pants - hat and ...",
      starting_price: 1000,
      imageSrc: "products/product1.svg",
      start_in: {
        days: 2,
        hours: 10,
        minutes: 50,
      },
      liked: false,
      tag: "live auction",
    },
    {
      title: "Six-piece clothing set (blouse - pants - hat and ...",
      starting_price: 1000,
      imageSrc: "products/product2.svg",
      start_in: {
        days: 2,
        hours: 10,
        minutes: 50,
      },
      liked: false,
      tag: "live auction",
    },
    {
      title: "Jeep Car, new, used for only one time",
      starting_price: 1000,
      imageSrc: "products/product3.svg",
      start_in: {
        days: 2,
        hours: 10,
        minutes: 50,
      },
      liked: true,
      tag: "hot sale",
    },
    {
      title: "Jeep Car, new, used for only one time",
      starting_price: 1000,
      imageSrc: "products/product4.svg",
      start_in: {
        days: 2,
        hours: 10,
        minutes: 50,
      },
      liked: false,
      tag: "live auction",
    },
    {
      title: "Six-piece clothing set (blouse - pants - hat and ...",
      starting_price: 1000,
      imageSrc: "products/product1.svg",
      start_in: {
        days: 2,
        hours: 10,
        minutes: 50,
      },
      liked: true,
      tag: "hot sale",
    },
  ],
}

export { profileData }
