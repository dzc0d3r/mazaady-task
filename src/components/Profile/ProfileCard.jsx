import Image from "next/image"
import { TbUserCheck, TbUsers } from "react-icons/tb"
import { RiMagicLine } from "react-icons/ri"
import { profileData } from "@/lib/data"

const ProfileCard = ({ fullname, bio, followers, following, rate }) => {
  return (
    <div className="mx-auto h-fit w-full max-w-full rounded-2xl bg-white p-5 md:max-w-none">
      <div className="flex flex-col">
        <Image
          src={profileData.imageSrc}
          height={100}
          width={100}
          alt="Profile image"
          className="h-24 w-24 rounded-2xl"
        ></Image>
      </div>
      <div className="mt-5">
        <h2 className="text-2xl font-semibold text-black/70">{fullname}</h2>
        <p className="mt-2 text-sm font-light text-black/70">{bio}</p>
      </div>
      <div className="my-5 flex flex-row gap-2">
        {/* stats */}
        <div className="flex w-full gap-1 rounded-2xl bg-orange-400 bg-opacity-10 p-2">
          {/* following */}
          <div className="flex items-center justify-center">
            <TbUserCheck className="text-2xl text-orange-400 md:text-3xl" />
          </div>
          <div className="flex flex-col">
            <p className="text-sm font-bold text-black/70 md:text-base">
              {following}
            </p>
            <span className="text-xs capitalize text-orange-400  md:text-sm">
              following
            </span>
          </div>
        </div>
        <div className="flex w-full gap-1 rounded-2xl bg-orange-400 bg-opacity-10 p-2">
          {/* followers */}
          <div className="flex items-center justify-center">
            <TbUsers className="text-2xl text-orange-400 md:text-3xl" />
          </div>
          <div className="flex flex-col">
            <p className="text-sm font-bold text-black/70 md:text-base">
              {followers}
            </p>
            <span className="text-xs capitalize text-orange-400  md:text-sm">
              followers
            </span>
          </div>
        </div>
        <div className="flex w-full gap-1 rounded-2xl bg-orange-400 bg-opacity-10 p-2">
          {/* following */}
          <div className="flex items-center justify-center">
            <RiMagicLine className="text-2xl font-semibold text-orange-400 md:text-3xl " />
          </div>
          <div className="flex flex-col">
            <p className="text-sm font-bold text-black/70 md:text-base">
              {rate.value}
              <span className="mx-2 text-xs font-light text-gray-600 md:text-sm">
                {rate.count}
              </span>
            </p>
            <span className="text-xs capitalize text-orange-400  md:text-sm">
              rate
            </span>
          </div>
        </div>
      </div>
      <button className="mt-5 flex w-full flex-row items-center justify-center gap-x-2 rounded-xl bg-gradient-to-r from-primary to-orange-600 px-4 py-2 text-center text-lg   text-white hover:to-orange-500 disabled:pointer-events-none disabled:opacity-50">
        Follow
      </button>
    </div>
  )
}

export default ProfileCard
