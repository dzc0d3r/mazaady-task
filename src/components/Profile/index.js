import ProfileCard from "./ProfileCard"
import QRCard from "./QRCard"
import Products from "./Products"
import SingleProduct from "./SingleProduct"

export { ProfileCard, QRCard, Products, SingleProduct }
