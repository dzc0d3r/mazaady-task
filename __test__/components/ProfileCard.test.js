import { render, screen } from "@testing-library/react"
import { ProfileCard } from "@/components/Profile"
import { profileData } from "@/lib/data"
import renderer from "react-test-renderer"

describe("ProfileCard", () => {
  const mockProps = { ...profileData }

  it("renders correctly", () => {
    const tree = renderer.create(<ProfileCard {...mockProps} />).toJSON()
    expect(tree).toMatchSnapshot()
  })
  it("renders the profile information", () => {
    render(<ProfileCard {...mockProps} />)
    expect(screen.getByText(`${mockProps.fullname}`)).toBeInTheDocument()
    expect(screen.getByText(`${mockProps.bio}`)).toBeInTheDocument()
    expect(screen.getByText(`${mockProps.followers}`)).toBeInTheDocument()
    expect(screen.getByText("followers")).toBeInTheDocument()
    expect(screen.getByText(`${mockProps.following}`)).toBeInTheDocument()
    expect(screen.getByText("following")).toBeInTheDocument()
    expect(screen.getByText(`${mockProps.rate.value}`)).toBeInTheDocument()
    expect(screen.getByText(`${mockProps.rate.count}`)).toBeInTheDocument()
  })

  it("renders the follow button", () => {
    render(<ProfileCard {...mockProps} />)
    expect(screen.getByText("Follow")).toBeInTheDocument()
  })
})
