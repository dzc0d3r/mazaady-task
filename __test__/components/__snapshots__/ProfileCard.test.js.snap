// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProfileCard renders correctly 1`] = `
<div
  className="h-fit w-full bg-white rounded-2xl p-5 max-w-full md:max-w-none mx-auto"
>
  <div
    className="flex flex-col"
  >
    <img
      alt="Profile image"
      className="rounded-2xl w-24 h-24"
      data-nimg="1"
      decoding="async"
      height={100}
      loading="lazy"
      onError={[Function]}
      onLoad={[Function]}
      src="profile-card.svg"
      style={
        {
          "color": "transparent",
        }
      }
      width={100}
    />
  </div>
  <div
    className="mt-5"
  >
    <h2
      className="text-2xl text-black/70 font-semibold"
    >
      Hala Ahmed
    </h2>
    <p
      className="mt-2 text-black/70 font-light text-sm"
    >
      I am Hala Ahmed, I am the owner of the local brand called Beauty which is for Mackeup and Skin Care.
    </p>
  </div>
  <div
    className="flex flex-row gap-2 my-5"
  >
    <div
      className="p-2 rounded-2xl bg-orange-400 bg-opacity-10 gap-1 flex w-full"
    >
      <div
        className="flex items-center justify-center"
      >
        <svg
          className="text-2xl md:text-3xl text-orange-400"
          fill="none"
          height="1em"
          stroke="currentColor"
          strokeLinecap="round"
          strokeLinejoin="round"
          strokeWidth="2"
          style={
            {
              "color": undefined,
            }
          }
          viewBox="0 0 24 24"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M0 0h24v24H0z"
            fill="none"
            stroke="none"
          />
          <path
            d="M8 7a4 4 0 1 0 8 0a4 4 0 0 0 -8 0"
          />
          <path
            d="M6 21v-2a4 4 0 0 1 4 -4h4"
          />
          <path
            d="M15 19l2 2l4 -4"
          />
        </svg>
      </div>
      <div
        className="flex flex-col"
      >
        <p
          className="text-sm md:text-base font-bold text-black/70"
        >
          10
        </p>
        <span
          className="text-xs md:text-sm capitalize  text-orange-400"
        >
          following
        </span>
      </div>
    </div>
    <div
      className="p-2 rounded-2xl bg-orange-400 bg-opacity-10 gap-1 flex w-full"
    >
      <div
        className="flex items-center justify-center"
      >
        <svg
          className="text-2xl md:text-3xl text-orange-400"
          fill="none"
          height="1em"
          stroke="currentColor"
          strokeLinecap="round"
          strokeLinejoin="round"
          strokeWidth="2"
          style={
            {
              "color": undefined,
            }
          }
          viewBox="0 0 24 24"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M0 0h24v24H0z"
            fill="none"
            stroke="none"
          />
          <path
            d="M9 7m-4 0a4 4 0 1 0 8 0a4 4 0 1 0 -8 0"
          />
          <path
            d="M3 21v-2a4 4 0 0 1 4 -4h4a4 4 0 0 1 4 4v2"
          />
          <path
            d="M16 3.13a4 4 0 0 1 0 7.75"
          />
          <path
            d="M21 21v-2a4 4 0 0 0 -3 -3.85"
          />
        </svg>
      </div>
      <div
        className="flex flex-col"
      >
        <p
          className="text-sm md:text-base font-bold text-black/70"
        >
          5
        </p>
        <span
          className="text-xs md:text-sm capitalize  text-orange-400"
        >
          followers
        </span>
      </div>
    </div>
    <div
      className="p-2 rounded-2xl bg-orange-400 bg-opacity-10 gap-1 flex w-full"
    >
      <div
        className="flex items-center justify-center"
      >
        <svg
          className="text-2xl md:text-3xl text-orange-400 font-semibold "
          fill="currentColor"
          height="1em"
          stroke="currentColor"
          strokeWidth="0"
          style={
            {
              "color": undefined,
            }
          }
          viewBox="0 0 24 24"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M15.1986 9.94435C14.7649 9.53358 14.4859 8.98601 14.4085 8.39371L14.0056 5.31126L11.275 6.79711C10.7503 7.08262 10.1433 7.17876 9.55608 7.06936L6.49998 6.50003L7.06931 9.55612C7.17871 10.1434 7.08257 10.7503 6.79707 11.275L5.31121 14.0056L8.39367 14.4085C8.98596 14.4859 9.53353 14.7649 9.94431 15.1986L12.0821 17.4555L13.4178 14.6485C13.6745 14.1091 14.109 13.6745 14.6484 13.4179L17.4555 12.0821L15.1986 9.94435ZM15.2238 15.5078L13.0111 20.1579C12.8687 20.4572 12.5107 20.5843 12.2115 20.4419C12.1448 20.4102 12.0845 20.3664 12.0337 20.3127L8.49229 16.574C8.39749 16.4739 8.27113 16.4095 8.13445 16.3917L3.02816 15.7242C2.69958 15.6812 2.46804 15.3801 2.51099 15.0515C2.52056 14.9782 2.54359 14.9074 2.5789 14.8425L5.04031 10.3191C5.1062 10.198 5.12839 10.0579 5.10314 9.92241L4.16 4.85979C4.09931 4.53402 4.3142 4.22074 4.63997 4.16005C4.7126 4.14652 4.78711 4.14652 4.85974 4.16005L9.92237 5.10319C10.0579 5.12843 10.198 5.10625 10.319 5.04036L14.8424 2.57895C15.1335 2.42056 15.4979 2.52812 15.6562 2.81919C15.6916 2.88409 15.7146 2.95495 15.7241 3.02821L16.3916 8.13449C16.4095 8.27118 16.4739 8.39754 16.5739 8.49233L20.3127 12.0337C20.5533 12.2616 20.5636 12.6414 20.3357 12.8819C20.2849 12.9356 20.2246 12.9794 20.1579 13.0111L15.5078 15.2238C15.3833 15.2831 15.283 15.3833 15.2238 15.5078ZM16.0206 17.4349L17.4348 16.0207L21.6775 20.2633L20.2633 21.6775L16.0206 17.4349Z"
          />
        </svg>
      </div>
      <div
        className="flex flex-col"
      >
        <p
          className="text-sm md:text-base font-bold text-black/70"
        >
          4.2
          <span
            className="text-xs md:text-sm text-gray-600 font-light mx-2"
          >
            15
          </span>
        </p>
        <span
          className="text-xs md:text-sm capitalize  text-orange-400"
        >
          rate
        </span>
      </div>
    </div>
  </div>
  <button
    className="w-full text-center justify-center mt-5 flex flex-row items-center gap-x-2 rounded-xl bg-gradient-to-r from-primary to-orange-600 px-4 py-2 text-lg   text-white hover:to-orange-500 disabled:pointer-events-none disabled:opacity-50"
  >
    Follow
  </button>
</div>
`;
